tests/unittests/tests-core/tests-core-xfa.c

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (C) 2018 Eistec AB
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include <stdint.h>
+#include "bitarithm.h"
+#include "xfa.h"
+
+#include "embUnit.h"
+
+#include "tests-core.h"
+#include "tests-core-xfa.h"
+
+XFA_INIT(xfatest_t, xfatest);
+XFA_INIT_CONST(xfatest_t, xfatest_const);
+XFA_USE(xfatest_t, xfatest_use);
+XFA_USE_CONST(xfatest_t, xfatest_use_const);
+
+/* Verifying that cross file array linking is correct by iterating over an external array */
+static void test_xfa_data(void)
+{
+    unsigned n = XFA_LEN(xfatest_t, xfatest);
+    TEST_ASSERT_EQUAL_INT(2, n);
+    unsigned found = 0;
+    for (unsigned k = 0; k < n; ++k) {
+        /* we do not want to enforce the order of the data elements */
+        switch (xfatest[k].val) {
+            case 12345:
+                /* tests-core-xfa-data1.c */
+                TEST_ASSERT_EQUAL_STRING("xfatest1", xfatest[k].text);
+                TEST_ASSERT(!(found & BIT0));
+                found |= BIT0;
+                break;
+            case 0xbeef:
+                /* tests-core-xfa-data2.c */
+                TEST_ASSERT_EQUAL_STRING("another test string", xfatest[k].text);
+                TEST_ASSERT(!(found & BIT1));
+                found |= BIT1;
+                break;
+            default:
+                break;
+        }
+    }
+    TEST_ASSERT_EQUAL_INT((1U << (n + 1)) - 1, found);
```

Comment:
```suggestion
    TEST_ASSERT_EQUAL_INT((1U << n) - 1, found);
```

Sorry, my bad. I should better leave calculations to the machines :-D